### PR TITLE
txm: add mass testing

### PR DESCRIPTION
**Pull request by Egor2001**
Test `mvcc_mass` verifies serializability and correctness of memtx with
mvcc engine trying to find equivalent sequenial order of transactions in memtx
with mvcc disabled. Transactions order and statements are generated randomly.

Needed for: #5999

### CODE REVIEW

**Comment by Egor2001** [comment]
It must become easier to debug with the #6226 feature as reproducers may generate `.dot`-dump of MVCC state before each `begin` and after each `commit`. 

<details>
<summary>Here is a `txn_proxy.lua` modified to help verify reproducers generated by the test</summary>

```lua
local ffi = require('ffi')
local yaml = require('yaml')
local json = require('json')
local fiber = require('fiber')
local console = require('console')

local array_mt = { __serialize = 'array' }

local mt = {
    __call = function(self, code_str)
        self.c1:put(code_str)
        local res = yaml.decode(self.c2:get())
        print(code_str..' => '..json.encode(res))
        return type(res) == 'table' and setmetatable(res, array_mt) or res
    end,
    __index = {
        begin    = function(self) return self('box.begin()') end,
        commit   = function(self) return self('box.commit()') end,
        rollback = function(self) return self('box.rollback()') end,
        close    = function(self) self.c1:close(); self.c2:close() end
    }
}

local function fiber_main(c1, c2)
    local code_str = c1:get()
    if code_str then
        c2:put(console.eval(code_str))
        return fiber_main(c1, c2) -- tail call
    end
end

local function new_txn_proxy()
    local c1, c2 = fiber.channel(), fiber.channel()
    local function on_gc() c1:close(); c2:close() end
    fiber.create(fiber_main, c1, c2)
    return setmetatable({
        c1 = c1, c2 = c2,
        __gc = ffi.gc(ffi.new('char[1]'), on_gc)
    }, mt)
end

return { new = new_txn_proxy }
```
</details>

File: test/box/mvcc_mass.test.lua [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+math.randomseed(os.time());
+
+-- Probability of the rollback
+rollback_prob = 0.05
+
+-- Probability of the commit
+commit_prob = 0.2
+
+-- Maximum random key value (min is 0)
+max_key_val = 100
```
Comment (alyapunov) [inline test/box/mvcc_mass.test.lua:38]: I would propose to set this to 10. I checked and saw that with 100 the majority of selects has empty results.

File: test/box/mvcc_mass.test.lua [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+conn:wait_connected()
+
+seq = box.schema.space.create('s', {engine='memtx', if_not_exists=true})
+seq_tree = seq:create_index('st', {type='tree'})
+seq_hash = seq:create_index('sh', {type='hash'})
+
+ntx, nro = 20, 8
+niters = 500
+stmts, ro_mask, ok_mask = conn:call('gen_statements', {ntx, nro, niters})
+result, what = try_serialize(stmts, ntx, ro_mask, ok_mask)
```
Comment (alyapunov) [inline test/box/mvcc_mass.test.lua:368]: I think we should do `gen_statements` + `try_serialize` in cycle, recreating spaces each time.

**Review by alyapunov — CHANGES_REQUESTED** [review]

File: test/box/mvcc_mass.test.lua (hunk 2 above)
Comment (Egor2001) [reply test/box/mvcc_mass.test.lua]: Fixed.

File: test/box/mvcc_mass.test.lua (hunk 1 above)
Comment (Egor2001) [reply test/box/mvcc_mass.test.lua]: Fixed.

File: test/box/mvcc_mass.test.lua [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+    txn.cancelled = true
+    return true
+end;
+
+function txn_commit(txn)
+    if not txn.running or txn.commited or txn.cancelled then
+        return error('cant commit txn')
+    end
+    local res = txn.handle:commit()
+    if res and res[1] and res[1]['error'] ==
```
[the remark this answers is not in the document]
Comment (Egor2001) [reply test/box/mvcc_mass.test.lua]: Fixed (and added assert to check that any other errors aren't possible in commit yet).

File: test/box/mvcc_mass.test.lua [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+        return error('cant begin txn')
+    end
+    txn.handle:begin()
+    txn.running = true
+    return true
+end;
+
+function txn_rollback(txn)
+    if not txn.running or txn.commited or txn.cancelled then
+        return error('cant rollback txn')
```
[the remark this answers is not in the document]
Comment (Egor2001) [reply test/box/mvcc_mass.test.lua]: Fixed hereinafter.

File: test/box/mvcc_mass.test.lua [hunk 5]
```diff
@@ -0,0 +1,423 @@
+test_run = require('test_run').new()
```
[the remark this answers is not in the document]
Comment (Egor2001) [reply test/box/mvcc_mass.test.lua]: Renamed.

File: test/box/mvcc_mass.test.lua [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+netbox = require('net.box')
+console = require('console')
+yaml = require('yaml')
+json = require('json')
+--------------------------------------------------------------------------------
+
+-- False iff only successful txns included in reprocucer
```
[the remark this answers is not in the document]
Comment (Egor2001) [reply test/box/mvcc_mass.test.lua]: Fixed ("iff" isn't a typo here).

File: test/box/mvcc_mass.lua [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env tarantool
+
+box.cfg{
+    listen                 = '127.0.0.1:3333',
```
[the remark this answers is not in the document]
Comment (Egor2001) [reply test/box/mvcc_mass.lua]: Thanks, fixed.

File: test/box/mvcc_mass.test.lua (hunk 6 above)
[the remark this answers is not in the document]
Comment (alyapunov) [reply test/box/mvcc_mass.test.lua]: iff - ok. I hope everybody will understand..

**Pull request closed without merging**